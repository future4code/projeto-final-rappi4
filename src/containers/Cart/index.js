import React, { Component } from 'react'
import { connect } from 'react-redux'
import {
    CartItensWrapper, NavBar, AppWrapper,
    AppBar, TitleAppBar, Frete, SubTotalWrapper,
    SubTotal, SubTotalValue, Payment, DividerStyled,
     ButtonStyled, StyledTitle, StyledAdress, StyledTime,
      StyledRestContent, ProductsWrapper, DeliveryAdress, DeliveryAdressClient, AdressWrapper
} from './style'
import ProductCard from '../../components/ProductCard'

import { FormControl, RadioGroup, FormControlLabel, Radio } from '@material-ui/core'

//TODO AJUSTAR STYLO DO RADIO
const FormControlControl = <Radio style={{padding:0, color:'black'}} />
const FormControlStyle = {marginLeft:'16px', marginBottom:'8px'}

export class Cart extends Component {
    constructor(props) {
        super(props)
        this.state = {
            selectedValue: '',
        }
    }
    handleChange = event => {
        this.setState({ selectedValue: event.target.value });
      };
    render() {
        
        return (
            <AppWrapper>
                <AppBar>
                    <TitleAppBar>Carrinho</TitleAppBar>
                </AppBar>
                <AdressWrapper>
                    <DeliveryAdress> Endereco de entrega</DeliveryAdress>
                    <DeliveryAdressClient>Endereco do cliente</DeliveryAdressClient>
                </AdressWrapper>
                <CartItensWrapper>
                    <StyledRestContent>
                        <StyledTitle>Restaurante bom</StyledTitle>
                        <StyledAdress>Rua 2, 344</StyledAdress>
                        <StyledTime>30 - 40</StyledTime>
                    </StyledRestContent>
                    <ProductsWrapper>
                    {/* <ProductCard/>
                    <ProductCard/> */}
                    </ProductsWrapper>
                </CartItensWrapper>
                <Frete>Frete R$10,00</Frete>
                <SubTotalWrapper>
                    <SubTotal>SUBTOTAL   </SubTotal>
                    <SubTotalValue>R$0,00</SubTotalValue>
                </SubTotalWrapper>
                <Payment>Forma de Pagamento</Payment>
                <DividerStyled />
                <FormControl component="fieldset" >
          <RadioGroup
            aria-label="Forma Pagamento"
            name="Forma Pagamento"
            value={this.state.value}
            onChange={this.handleChange}
          >
            <FormControlLabel
              value="Dinheiro"
              control={FormControlControl}
              label="Dinheiro"
              style={FormControlStyle}
            />
            <FormControlLabel
              value="Cartão de Crédito"
              control={FormControlControl}
              label="Cartão de Crédito"
              style={FormControlStyle}
            />
          </RadioGroup>
        </FormControl>
                <ButtonStyled>Confirmar</ButtonStyled>
                <NavBar>
                    Aqui vai a nav Bar, fixa no final
                </NavBar>
            </AppWrapper>
        )
    }
}

// const mapStateToProps = (state) => ({

// })

// const mapDispatchToProps = {

// }

// export default connect(mapStateToProps, mapDispatchToProps)(Cart)
export default Cart
