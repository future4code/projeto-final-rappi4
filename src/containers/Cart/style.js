import styled from 'styled-components'
import Divider from '@material-ui/core/Divider';
import { withStyles } from '@material-ui/core/styles';
import Radio from '@material-ui/core/Radio';
import { FormControl, FormControlLabel } from '@material-ui/core';

export const AppWrapper = styled.div`
    display:flex;
    flex-direction:column;
    height: 600px;
    font-family: Roboto;
`

export const CartItensWrapper = styled.div`
    display:flex;
    align-items:center;
    align-self:center;
    flex-direction:column;
    /* padding:0 16px; */
`

export const NavBar = styled.div`
    height: 49px;
`

export const Frete = styled.div`
    width: 104px;
    height: 18px;
    font-family: Roboto;
    font-size: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    text-align: right;
    color: var(--black);
    align-self:flex-end;
    padding-right:16px;
    margin-bottom:14px;
    margin-top:8px;
`

export const TotalWrapper = styled.div`
    display:flex;
    flex-direction:column;
    align-items:flex-end;
    margin-bottom:24px;
`

export const SubTotalWrapper = styled.div`
    display:flex;
    justify-content: space-between;
    padding: 0 16px;
    margin-bottom:25px;
`

export const Payment = styled.div`
    width: 360px;
    height: 18px;
    font-family: Roboto;
    font-size: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    color: var(--black);
    padding: 0 16px;
    margin-bottom:8px;
`

export const SubTotal = styled.div`
    width: 164px;
    height: 18px;
    font-family: Roboto;
    font-size: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    color: var(--black);
`

export const SubTotalValue = styled.div`
    width: 164px;
    height: 21px;
    font-family: Roboto;
    font-size: 18px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.43px;
    text-align: right;
    color: #e86e5a;
`

export const DividerStyled = styled(Divider)`
    margin: 0 16px 8px 16px;
    background-color:black; 
`

export const ButtonStyled = styled.button`
    height: 42px;
    border-radius: 2px;
    background-color: #e86e5a;
    align-self:center;
    margin:0;
    border:0;
    width:100vw;
`
export const StyledRestContent = styled.div`
    padding: 16px;
    width:100vw;
`
export const ProductsWrapper = styled.div`
    padding: 0 16px;
    display:flex;
    flex-direction:column;
`

export const StyledTitle = styled.div`
    display: block;
    color: #e86e5a;
    font-size: 16px;
    margin-bottom: 8px; 
`
export const StyledAdress = styled.div`
    display: block;
    font-size: 16px;
    color: #b8b8b8;
    margin-bottom: 8px;
`
export const StyledTime = styled.div`
    display: block;
    font-size: 16px;
    color: #b8b8b8;
`

export const DeliveryAdress = styled.div`
    width: 328px;
    height: 18px;
    font-family: Roboto;
    font-size: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    color: #b8b8b8;
    margin-bottom:8px;
`

export const DeliveryAdressClient = styled.div`
    width: 328px;
    height: 18px;
    font-family: Roboto;
    font-size: 16px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: -0.39px;
    color: var(--black);
`

export const AdressWrapper = styled.div`
    background-color: #eeeeee;
    padding:16px;
`

export const StyledRadio = withStyles({
  root: {
    color:"black",
    padding:"0",
  },
  
  
})(Radio);

export const StyledFormControl = withStyles({
  root: {
    margin:'0 16px',
  },
})(FormControl);

export const FormControlLabelSyled = withStyles({
  root: {
    margin:'0',
    marginBottom:'8px'
  },
  label: {
    marginLeft: '8px',
  },
})(FormControlLabel);