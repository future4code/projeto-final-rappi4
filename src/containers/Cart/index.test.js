import React from 'react';
import renderer from 'react-test-renderer';
import Cart from '.';

describe("Cart Container ",()=>{
    it("should render ",()=>{
        const tree = renderer.create(<Cart/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})