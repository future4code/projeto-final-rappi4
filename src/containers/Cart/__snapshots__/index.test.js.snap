// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Container  should render  1`] = `
<div
  className="sc-bdVaJa hknDB"
>
  <div
    className="sc-bwzfXH lgtiFH"
  >
    <div
      className="sc-htpNat ccTSqx"
    >
      Carrinho
    </div>
  </div>
  <div
    className="sc-kgoBCf cnRrMq"
  >
    <div
      className="sc-kAzzGY fMzrZJ"
    >
       Endereco de entrega
    </div>
    <div
      className="sc-chPdSV hrnCLJ"
    >
      Endereco do cliente
    </div>
  </div>
  <div
    className="sc-bxivhb jZFhqI"
  >
    <div
      className="sc-VigVT fxjpfs"
    >
      <div
        className="sc-fjdhpX ewHUQA"
      >
        Restaurante bom
      </div>
      <div
        className="sc-jzJRlG gdnCLB"
      >
        Rua 2, 344
      </div>
      <div
        className="sc-cSHVUG kxuaIn"
      >
        30 - 40
      </div>
    </div>
    <div
      className="sc-jTzLTM hYfSKd"
    />
  </div>
  <div
    className="sc-EHOje iKhxjT"
  >
    Frete R$10,00
  </div>
  <div
    className="sc-gzVnrw gVVMKi"
  >
    <div
      className="sc-dnqmqq cCkgVs"
    >
      SUBTOTAL   
    </div>
    <div
      className="sc-iwsKbI cdoMaD"
    >
      R$0,00
    </div>
  </div>
  <div
    className="sc-htoDjs jwFhsk"
  >
    Forma de Pagamento
  </div>
  <hr
    className="MuiDivider-root-1 sc-gZMcBi zDdSs"
  />
  <fieldset
    className="MuiFormControl-root-6"
  >
    <div
      aria-label="Forma Pagamento"
      className="MuiFormGroup-root-10"
      role="radiogroup"
    >
      <label
        className="MuiFormControlLabel-root-12"
        style={
          Object {
            "marginBottom": "8px",
            "marginLeft": "16px",
          }
        }
      >
        <span
          className="MuiButtonBase-root-33 MuiIconButton-root-27 MuiPrivateSwitchBase-root-23 MuiRadio-root-18 MuiRadio-colorSecondary-22"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "color": "black",
              "padding": 0,
            }
          }
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-32"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-36"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <input
              checked={false}
              className="MuiPrivateSwitchBase-input-26"
              disabled={false}
              name="Forma Pagamento"
              onChange={[Function]}
              type="radio"
              value="Dinheiro"
            />
          </span>
          <span
            className="MuiTouchRipple-root-81"
          />
        </span>
        <span
          className="MuiTypography-root-45 MuiTypography-body1-54 MuiFormControlLabel-label-17"
        >
          Dinheiro
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root-12"
        style={
          Object {
            "marginBottom": "8px",
            "marginLeft": "16px",
          }
        }
      >
        <span
          className="MuiButtonBase-root-33 MuiIconButton-root-27 MuiPrivateSwitchBase-root-23 MuiRadio-root-18 MuiRadio-colorSecondary-22"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "color": "black",
              "padding": 0,
            }
          }
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-32"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-36"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <input
              checked={false}
              className="MuiPrivateSwitchBase-input-26"
              disabled={false}
              name="Forma Pagamento"
              onChange={[Function]}
              type="radio"
              value="Cartão de Crédito"
            />
          </span>
          <span
            className="MuiTouchRipple-root-81"
          />
        </span>
        <span
          className="MuiTypography-root-45 MuiTypography-body1-54 MuiFormControlLabel-label-17"
        >
          Cartão de Crédito
        </span>
      </label>
    </div>
  </fieldset>
  <button
    className="sc-gqjmRU gELACb"
  >
    Confirmar
  </button>
  <div
    className="sc-ifAKCX cktfmg"
  >
    Aqui vai a nav Bar, fixa no final
  </div>
</div>
`;
